eth/api_backend.go

```diff
@@ -17,189 +17,299 @@
 package eth
 
 import (
+	"errors"
+	"fmt"
 	"math/big"
 
+	ethereum "github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/state"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/core/vm"
-	"github.com/ethereum/go-ethereum/eth/downloader"
-	"github.com/ethereum/go-ethereum/eth/gasprice"
-	"github.com/ethereum/go-ethereum/ethdb"
-	"github.com/ethereum/go-ethereum/event"
 	"github.com/ethereum/go-ethereum/internal/ethapi"
-	rpc "github.com/ethereum/go-ethereum/rpc"
 	"golang.org/x/net/context"
 )
 
-// EthApiBackend implements ethapi.Backend for full nodes
-type EthApiBackend struct {
-	eth *Ethereum
-	gpo *gasprice.GasPriceOracle
+// Assert at compile time that this implementation has all optional API features.
+var (
+	_ ethapi.TransactionInclusionBlock = (*Ethereum)(nil)
+	_ ethapi.PendingState              = (*Ethereum)(nil)
+	_ bind.ContractBackend             = (*Ethereum)(nil)
+	_ bind.PendingContractCaller       = (*Ethereum)(nil)
+)
+
+var (
+	ErrBlockNotFound = errors.New("block not found")
+	ErrTxNotFound    = errors.New("transaction not found")
+)
```

Comment:
I'd fix this in internal/ethapi, but let Go API return an error instead. Do you agree?